scdl/scdl.py

```diff
@@ -458,64 +463,119 @@ def download_track(track, playlist_name=None, playlist_file=None):
+    total_length = int(r.headers.get('content-length'))
 
-        min_size = arguments.get('--min-size')
-        max_size = arguments.get('--max-size')
+    min_size = arguments.get('--min-size')
+    max_size = arguments.get('--max-size')
 
-        if min_size is not None and total_length < min_size:
-            logging.info('{0} not large enough, skipping'.format(title))
-            return
+    if min_size is not None and total_length < min_size:
+        logging.info('{0} not large enough, skipping'.format(title))
+        return
 
-        if max_size is not None and total_length > max_size:
-            logging.info('{0} too large, skipping'.format(title))
-            return
+    if max_size is not None and total_length > max_size:
+        logging.info('{0} too large, skipping'.format(title))
+        return
 
-        with temp as f:
-            for chunk in progress.bar(
-                r.iter_content(chunk_size=1024),
-                expected_size=(total_length/1024) + 1,
-                hide=True if arguments["--hide-progress"] else False
-            ):
-                if chunk:
-                    f.write(chunk)
-                    f.flush()
-
-        shutil.move(temp.name, os.path.join(os.getcwd(), filename))
-        if filename.endswith('.mp3') or filename.endswith('.m4a'):
-            try:
-                set_metadata(track, filename, playlist_name)
-            except Exception as e:
-                logger.error('Error trying to set the tags...')
-                logger.debug(e)
-        else:
-            logger.error("This type of audio doesn't support tagging...")
+    with temp as f:
+        for chunk in progress.bar(
+            r.iter_content(chunk_size=1024),
+            expected_size=(total_length/1024) + 1,
+            hide=True if arguments["--hide-progress"] else False
+        ):
+            if chunk:
+                f.write(chunk)
+                f.flush()
+
+    shutil.move(temp.name, os.path.join(os.getcwd(), filename))
+    if filename.endswith('.mp3') or filename.endswith('.m4a'):
+        try:
+            set_metadata(track, filename, playlist_name)
+        except Exception as e:
+            logger.error('Error trying to set the tags...')
+            logger.debug(e)
+    else:
+        logger.error("This type of audio doesn't support tagging...")
 
-        #Try to change the real creation date
-        created_at = track['created_at']
-        filetime = int(time.mktime(datetime.strptime(created_at, '%Y/%m/%d %H:%M:%S %z').timetuple()))
-        try_utime(filename,filetime)
+    #Try to change the real creation date
+    created_at = track['created_at']
+    filetime = int(time.mktime(datetime.strptime(created_at, '%Y/%m/%d %H:%M:%S %z').timetuple()))
+    try_utime(filename,filetime)
 
-    else:
+    logger.info('{0} Downloaded.\n'.format(filename))
+    record_download_archive(track)
+
+
+def already_downloaded(track, title, filename=None):
```

Comment:
The description of the method does not correspond to its name, it is true if the file should be downloaded, so when it's NOT already downloaded.

Reply:
Good catch, thank you! Fixed now :)